docs/cookbook/rag.md

```diff
@@ -0,0 +1,257 @@
+# Retrieval Augmented Generation (RAG)
+
+When you need to provide additional information to your model that it hasn't been pretrained on, we retrieve the relevant information from an external source and provide it as context to our model.
```

Comment:
rather than summarize what rag is, i think this section should be a short one or two sentences about Mirascope RAG.

terrible e.g. For this recipe, we will demonstrate how easy it is to integrate RAG into Mirascope prompts.

thoughts? @willbakst @brianpark0126

Reply:
How about this:

```markdown
Often you want your model to generate its response based on relevant
information that you provide. This recipe will demonstrate how to retrieve
the relevant information and how easy it is to format that information into
a Mirascope prompt.
```

The above has newlines that should be removed when added, but I added them here so it's easier to read.